msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/msalonly/ltw/TestCase2582288.java

```diff
@@ -0,0 +1,130 @@
+import com.microsoft.identity.client.ui.automation.app.OneAuthTestApp;
+import com.microsoft.identity.client.ui.automation.broker.BrokerLTW;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.interaction.FirstPartyAppPromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+// If LTW is the active broker, and request is made through Authenticator from OneAuth, nothing should break
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2582288
+@LTWTests
+@RunOnAPI29Minus("OneAuth test app dependency")
+@SupportedBrokers(brokers = {BrokerLTW.class})
+public class TestCase2582288 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_2582288() throws Throwable {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        // Install new Authenticator app with broker SDK changes of broker selection logic
+        final BrokerMicrosoftAuthenticator brokerMicrosoftAuthenticator = new BrokerMicrosoftAuthenticator();
+        brokerMicrosoftAuthenticator.uninstall();
+        brokerMicrosoftAuthenticator.install();
+
+        // Install old MsalTestApp
+        final MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.uninstall();
+        msalTestApp.installOldApk();
+        msalTestApp.launch();
+        msalTestApp.handleFirstRun();
+
+        // AcquireToken Interactively in MsalTestApp
+        final FirstPartyAppPromptHandlerParameters promptHandlerParameters = FirstPartyAppPromptHandlerParameters.builder()
+                .broker(mBroker)
+                .prompt(PromptParameter.LOGIN)
+                .loginHint(username)
+                .consentPageExpected(false)
+                .speedBumpExpected(false)
+                .sessionExpected(false)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingLoginPageAccountPicker(false)
+                .enrollPageExpected(false)
+                .build();
+
+        final String tokenMsal = msalTestApp.acquireToken(username, password, promptHandlerParameters, true);
+        Assert.assertNotNull(tokenMsal);
+
+        // Install old OneAuthTestApp
+        final OneAuthTestApp oneAuthTestApp = new OneAuthTestApp();
+        oneAuthTestApp.uninstall();
+        oneAuthTestApp.installOldApk();
+        oneAuthTestApp.launch();
+        oneAuthTestApp.handleFirstRun();
+
+        // Enter username in AccountName textbox
+        oneAuthTestApp.handleUserNameInput(username);
+        oneAuthTestApp.selectFromAppConfiguration("com.microsoft.identity.LabsApi.Guest");
+        oneAuthTestApp.handlePreferBrokerSwitchButton();

```

Comment:
Are these two lines needed? Don't see them in the test steps

Reply:
Yes, Sowmya and I confirmed that the desired result can only be obtained by conducting these steps